server/src/modules/post/post.service.ts

```diff
@@ -343,20 +344,12 @@ export class PostService {
           'answer_count',
         ],
       ],
-    })
+    })) as PostWithUserRelations
```

Comment:
There is little need to do a cast here, since `post` can be considered a subtype of PostWithUserRelations.

See Liskov's Principle of Substitution.